akka-projection-slick/src/main/scala/akka/projection/slick/OffsetStore.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick
+
+import java.util.UUID
+
+import akka.Done
+import akka.annotation.ApiMayChange
+import slick.jdbc.JdbcProfile
+import akka.persistence.query
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@ApiMayChange
+class OffsetStore[P <: JdbcProfile](db: P#Backend#Database, profile: P) {
+
+  import profile.api._
+
+  // offset manifest
+  private val StringM = "STR"
+  private val LongM = "LNG"
+  private val IntM = "INT"
+  private val SequenceM = "SEQ"
+  private val TimeBasedUUIDM = "TBU"
+
+  def readOffset[Offset](projectionId: String)(implicit ec: ExecutionContext): Future[Option[Offset]] = {
```

Comment:
I think this is the name of the projection.  We should keep the terminology consistent in the implementations and tests.

```suggestion
  def readOffset[Offset](name: String)(implicit ec: ExecutionContext): Future[Option[Offset]] = {
```

Reply:
I will introduce a `ProjectionId` holding the `name` and the `key` instead. Similar to Akka's `PersistenceId`.

Then we can use if in the offset-store instead of keep concatenating those two.